packages/adblocker/src/engine/engine.ts

```diff
@@ -86,16 +86,29 @@ export interface Caching {
   write: (path: string, buffer: Uint8Array) => Promise<void>;
 }
 
-export default class FilterEngine extends EventEmitter<
-  | 'csp-injected'
-  | 'html-filtered'
-  | 'request-allowed'
-  | 'request-blocked'
-  | 'request-redirected'
-  | 'request-whitelisted'
-  | 'script-injected'
-  | 'style-injected'
-> {
+export type EngineEventContext<T> = {
+  hostname: string;
+  classes: string[] | undefined;
+  hrefs: string[] | undefined;
+  ids: string[] | undefined;
+  context?: T;
```

Comment:
maybe `any` would be sufficient

Reply:
I was also thinking about that. It'll simplify a lot but I think the type complexity is still handlable at this level. I'll keep this in mind.